Clicking an un-herped comment re-herps it

atomheartother opened the pull request:
Because sometimes you regret un-herpderping.

I've been wanting to see this feature for a bit, so here it is.

I'd rather not save the herped string for a simple reason: space. If we're saving a string that's a few words long for every comment on a youtube page we could significantly increase the space taken up in memory when loading a *lot* of comments. So instead we generate a new string everytime.
 
Another solution (to keep the herped string the same) would be to stop generating a random string and to instead use the base comment, hash it in some way and to use that hash to generate a herped comment. But that's too much a departure from the original algorithm, so unless you want me to do it this way I won't.

twstokes commented:
I like the idea! Can we use `textContent` over `innerHTML`? There are some security concerns with `innerHTML`, see #7.

atomheartother commented:
@twstokes I would assume that's possible, `innerHTML` is how the Safari version currently does things and I simply didn't want to change it. I have no mac computer though so I don't think I'm the right person to make that change.

twstokes commented:
Ok, no worries. I updated it not too long ago to not use `innerHTML`, so the Safari version needs to be updated. I may have time in the near future to check this out. Thanks.

atomheartother commented:
Oh I just realized I'd left an innerHTML in the firefox+chrome code, I thought you meant the changes I'd done to the Safari version, sillly me...

atomheartother commented:
@twstokes I have updated this branch further and changed the name of my pull request.

This branch now also attempts to fix both #12 and #13.
#13 :
- All global variables have been removed
- Functions are now declared with `function` and not `var func = function`
- The method that we use to derp comment is completelly changed. No more indefinite timeout, as long as the comments are loaded, we attach an observer to that DOM elements, and then the herp function is only called when the DOM is modified.

#12 
Because I was rewriting how we herp comments anyway, I decided to try and tackle this bug. It's not perfect. All I do is treat the symptoms, not fix the underlying issue: I take the messed up comment and undo the bad things.
This works, but it messes up on some large comments with line breaks in them. Still, it only cuts out part of the comment, it doesn't display a comment from a completely different video, which is what the current system does.
